pytorch/inference/docker/2.0/py3/Dockerfile.cpu

```diff
@@ -0,0 +1,247 @@
+    LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/usr/local/lib" \
+    LD_LIBRARY_PATH="/opt/conda/lib:${LD_LIBRARY_PATH}" \
+    PATH=/opt/conda/bin:$PATH \
+    TEMP=/home/model-server/tmp \
+    MKL_THREADING_LAYER=GNU \
+    DLC_CONTAINER_TYPE=inference \
+    # See http://bugs.python.org/issue19846
+    LANG=C.UTF-8
+
+RUN apt-get update \
+ && apt-get -y upgrade \
+ && apt-get install -y --no-install-recommends \
+    build-essential \
+    ca-certificates \
+    cmake \
+    curl \
+    emacs \
+    git \
+    jq \
+    libcurl4-openssl-dev \
+    libgl1-mesa-glx \
+    libglib2.0-0 \
+    libsm6 \
+    libssl-dev \
+    libxext6 \
+    libxrender-dev \
+    openjdk-17-jdk \
+    openssl \
+    unzip \
+    vim \
+    wget \
+    libjpeg-dev \
+    libpng-dev \
+    zlib1g-dev \
+ && apt-get autoremove -y \
+ && rm -rf /var/lib/apt/lists/* \
+ && apt-get clean
+
+# Install OpenMPI
+RUN wget https://download.open-mpi.org/release/open-mpi/v4.1/openmpi-${OPEN_MPI_VERSION}.tar.gz \
+ && gunzip -c openmpi-$OPEN_MPI_VERSION.tar.gz | tar xf - \
+ && cd openmpi-$OPEN_MPI_VERSION \
+ && ./configure --prefix=/home/.openmpi \
+ && make all install \
+ && cd .. \
+ && rm openmpi-$OPEN_MPI_VERSION.tar.gz \
+ && rm -rf openmpi-$OPEN_MPI_VERSION
+
+# The ENV variables declared below are changed in the previous section
+# Grouping these ENV variables in the first section causes
+# ompi_info to fail. This is only observed in CPU containers
+ENV PATH="$PATH:/home/.openmpi/bin"
+ENV LD_LIBRARY_PATH="$LD_LIBRARY_PATH:/home/.openmpi/lib/"
+RUN ompi_info --parsable --all | grep mpi_built_with_cuda_support:value
+
+# Install CondaForge miniconda
+RUN curl -L -o ~/mambaforge.sh https://github.com/conda-forge/miniforge/releases/download/${MAMBA_VERSION}/Mambaforge-${MAMBA_VERSION}-Linux-x86_64.sh \
+ && chmod +x ~/mambaforge.sh \
+ && ~/mambaforge.sh -b -p /opt/conda \
+ && rm ~/mambaforge.sh \
+ && /opt/conda/bin/conda install -c conda-forge python=${PYTHON_VERSION} \
+    cython \
+    mkl \
+    mkl-include \
+    parso \
+    typing \
+    h5py \
+    requests \
+    # Below 2 are included in miniconda base, but not mamba so need to install
+    conda-content-trust \
+    charset-normalizer \
+ && /opt/conda/bin/conda clean -ya
+
+# symlink pip for OS use
+RUN pip install --upgrade pip --trusted-host pypi.org --trusted-host files.pythonhosted.org \
+ && ln -s /opt/conda/bin/pip /usr/local/bin/pip3
+
+# Ensure PyTorch did not get installed from Conda
+RUN pip uninstall -y torch torchvision torchaudio torchdata model-archiver multi-model-server
+
+# Install AWS-PyTorch, and other torch packages
+RUN pip install --no-cache-dir --extra-index-url https://download.pytorch.org/whl/cpu -U \
+    boto3 \
+    scipy \
+    opencv-python \
+    cryptography \
+    "ipython>=8.10.0,<9.0" \
+    "awscli<2" \
+    # TS tsquirements 
+    # -> Numpy is higer version as required by torch
+    # -> Pillow is not upper pinned for security
+    psutil \
```

Comment:
Any reason to install TS requirements individually instead of directly from their repo like we do in PT 1.11 image?

Reply:
updating this